qiskit/backends/local/localjob.py

```diff
@@ -55,6 +59,20 @@ def __init__(self, fn, qobj):
         self._backend_name = qobj.header.backend_name
         self._future = None
 
+        # verify the QObj is valid before making requests for computing resources
+        sdk = qiskit_path[0]
+        # Schemas path:     qiskit/backends/schemas
+        schemas_path = os.path.join(sdk, 'schemas')
+        schema_file_path = os.path.join(schemas_path, 'qobj_schema.json')
+
+        with open(schema_file_path, 'r') as schema_file:
+            schema = json.load(schema_file)
+
+        try:
+            jsonschema.validate(self._qobj.as_dict(), schema)
```

Comment:
It seems like the validation should happen in the same class object for both `ibmq` and `local` backends. Now it seems to be happening in `ibmqbackend` and `localjob`.

Reply:
Perhaps it makes sense in BaseJob derived classes so in ibmqjob.py validation could be done in the `else` clause of this statement;
https://github.com/Qiskit/qiskit-terra/blob/d11bdd2523aba4b8f49d4fc8cf927a288b19cf3b/qiskit/backends/ibmq/ibmqjob.py#L139-L148

Reply:
I think yours is the better suggestion.

Reply:
We are still bringing up a device backend which can understand qobj 1.0. I'm not sure if it should be done then or when all devices have been updated.